dimod/discrete/cydiscrete_quadratic_model.pyx

```diff
@@ -83,38 +67,19 @@ cdef class cyDiscreteQuadraticModel:
         cdef Py_ssize_t num_samples = samples.shape[0]
         cdef VarIndex num_variables = samples.shape[1]
 
-        cdef Bias[:] energies = np.zeros(num_samples, dtype=self.dtype)
+        cdef Bias[:] energies = np.empty(num_samples, dtype=self.dtype)
```

Comment:
I do not need the array to be filled with zeroes, as I overwrite the values in the way I calculated energy.